cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/WorkflowBackedActionContext.java

```diff
@@ -52,4 +54,9 @@ public WorkflowToken getToken() {
   public boolean isSuccessful() {
     return workflowContext.getState().getStatus() == ProgramStatus.COMPLETED;
   }
+
+  @Override
+  public void record(List<Operation> operations) {
+    throw new UnsupportedOperationException("Lineage recording is not supported.");
```

Comment:
I didn't add it for the action plugins since they represent the control flow and not the data flow. There is no input/output schema associated with it. We can support recording lineage in Action plugins however this means it should record the complete lineage graph (which includes READ, TRANSFORM, and WRITE operations) right?

Reply:
I was thinking more of a case where- say - an action plugin makes an RPC call to populate a field in the workflow token, and a follow-on transform adds that field to every record. How would we represent something like this if action plugins cannot emit operations?

Reply:
I see. this use case make sense. In this case Action plugin can simply record Read operation which outputs some fields which are used by later transform. Have filed JIRA for this issue - https://issues.cask.co/browse/CDAP-13671. Change is simple but would need unit test case. will do in separate PR.